src/main/java/com/myFirstProject/myFirstProject/service/AccountServiceImpl.java

```diff
@@ -166,12 +168,30 @@ private BigDecimal convertSum(List<CurrencyRate> currencyRates, CurrencyEntity c
 
     @Override
     @Transactional
-    public void payForArticles(BigDecimal tacks, Long id) {
+    public void payForArticles(BigDecimal sum, Long id, PromoCode promoCode) {
         Account account = accountRepository.findAccountByUserId(id);
-        checkAccount(account, tacks);
-        BigDecimal subtract = account.getSum().subtract(tacks);
+        checkAccount(account, sum);
+        BigDecimal taxPlusDiscount = findSumOfOrder(promoCode, sum);
+        BigDecimal subtract = account.getSum().subtract(taxPlusDiscount);
         account.setSum(subtract);
-        logger.info(String.format("User with id - %d payed - %s, his account - %s", id, tacks, subtract));
+        logger.info(String.format("User with id - %d payed - %s, his account - %s", id, sum, subtract));
+    }
+
+    private BigDecimal findSumOfOrder(PromoCode promoCode, BigDecimal tacks) {
```

Comment:
again what is this mean? "tacks" - extremely strange name for BigDecimal

Reply:
+